src/Raven.Client/Documents/Conventions/JsonNetBlittableEntitySerializer.cs

```diff
@@ -25,6 +25,11 @@ public JsonNetBlittableEntitySerializer(DocumentConventions conventions)
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(conventions, null);
         }
 
+        public static void CleanThreadStatics()
```

Comment:
I don't particularly like this approach. It means that we spread the knowledge about clearing thread local state far and wide. We need to register this as an event, instead.